src/hotspot/share/runtime/thread.cpp

```diff
@@ -3863,13 +3863,7 @@ void Threads::print_on(outputStream* st, bool print_stacks,
   }
 
   PrintOnClosure cl(st);
-  cl.do_thread(VMThread::vm_thread());
-  Universe::heap()->gc_threads_do(&cl);
-  if (StringDedup::is_enabled()) {
-    StringDedup::threads_do(&cl);
-  }
-  cl.do_thread(WatcherThread::watcher_thread());
-  cl.do_thread(AsyncLogWriter::instance());
+  non_java_threads_do(&cl);
 
   // SapMachine 2019-11-07: Vitals
```

Comment:
I think L3868 - L3883 can be also removed. The vitals sampler thread and the himem reporter thread are `NonJavaThread`s and handled by `non_java_threads_do(&cl);` in L3866.

Reply:
Interesting, good catch. Then it's probably also wrong in head and the other higher backports, e.g.: https://github.com/SAP/SapMachine/blob/32cfd1865848364f004da243636a3951c188754a/src/hotspot/share/runtime/threads.cpp#L1389 correct?

Reply:
Yes indeed. Likewise good catch.